jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverter.java

```diff
@@ -691,63 +674,58 @@ private static ObjectNode transformNodeLL(JsonNode oldNode) {
-      // 15 bit value
-      return "node-LL3";
-    } else if (((transformedLatabs & (-1 << 17)) == 0
-        || (transformedLat < 0 && (transformedLatabs ^ (1 << 17)) == 0))
-        && (transformedLonabs & (-1 << 17)) == 0
-        || (transformedLon < 0 && ((transformedLonabs ^ (1 << 17)) == 0))) {
-      // 17 bit value
-      return "node-LL4";
-    } else if (((transformedLatabs & (-1 << 21)) == 0
-        || (transformedLat < 0 && (transformedLatabs ^ (1 << 21)) == 0))
-        && (transformedLonabs & (-1 << 21)) == 0
-        || (transformedLon < 0 && ((transformedLonabs ^ (1 << 21)) == 0))) {
-      // 21 bit value
-      return "node-LL5";
-    } else if (((transformedLatabs & (-1 << 23)) == 0
-        || (transformedLat < 0 && (transformedLatabs ^ (1 << 23)) == 0))
-        && (transformedLonabs & (-1 << 23)) == 0
-        || (transformedLon < 0 && ((transformedLonabs ^ (1 << 23)) == 0))) {
-      // 23 bit value
-      return "node-LL6";
+  public static final String NODE_LL1 = "node-LL1"; // Node-LL-24B, within ±22.634554 meters of last node
+  public static final String NODE_LL2 = "node-LL2"; // Node-LL-28B, within ±90.571389 meters of last node
+  public static final String NODE_LL3 = "node-LL3"; // Node-LL-32B, within ±362.31873 meters of last node
+  public static final String NODE_LL4 = "node-LL4"; // Node-LL-36B, within ±1.449308 kilometers of last node
+  public static final String NODE_LL5 = "node-LL5"; // Node-LL-44B, within ±23.189096 kilometers of last node
+  public static final String NODE_LL6 = "node-LL6"; // Node-LL-48B, within ±92.756481 kilometers of last node
+
+  // -- Limits for each node type
+  public final static long NODE_LL1_LIMIT = 2047;
+  public final static long NODE_LL2_LIMIT = 8191;
+  public final static long NODE_LL3_LIMIT = 32767;
+  public final static long NODE_LL4_LIMIT = 131071;
+  public final static long NODE_LL5_LIMIT = 2097151;
+  public final static long NODE_LL6_LIMIT = 8388607;
+  // In J2735, the value -8388608 indicates an unknown value and is considered invalid because it falls outside the acceptable ± range.
+
+  /**
+   * Determines the node offset point LL type based on the latitude and longitude deltas.
+   * The method evaluates the absolute values of the deltas against predefined limits to determine
+   * the appropriate node type. If the deltas do not fit within any allowed ranges, an exception
+   * is thrown.
+   *
+   * @param latDelta The latitude delta as a long value.
+   * @param lonDelta The longitude delta as a long value.
+   * @return A string representing the node offset point LL type (e.g., NODE_LL1, NODE_LL2, etc.).
+   * @throws InvalidNodeLatLonOffsetException if latDelta or lonDelta are outside the permissible range
+   *                                          of -0.8388608 to +0.8388607 degrees.
+   */
+  public static String determineNodeOffsetPointLLType(long latDelta, long lonDelta) throws InvalidNodeLatLonOffsetException {

```

Comment:
issue: this method is not intended for use outside of this class. As such, it should not be marked as public.

suggestion(blocking): since we're testing this method directly (good choice, by the way), we can mark it as protected to allow for tests to access the method, while restricting access to this method.
```suggestion
  protected static String determineNodeOffsetPointLLType(long latDelta, long lonDelta) throws InvalidNodeLatLonOffsetException {

```

Reply:
Makes sense, this in now `protected`!